tests.rb

```diff
@@ -1,28 +1,314 @@
+    assert_equal course.id, lesson.course_id
+  end
+
+  def test_courses_instructors_association
+    user = User.create(first_name: "Test", last_name: "User", email: "test@used.com", instructor: true)
+    course = Course.create(name: "Math", course_code: "AAA1544311")
+    course_instructor = CourseInstructor.create(course_id: course.id, instructor_id: user.id)
+    refute course.destroy
+    assert_equal course_instructor.course_id, course.id
+  end
+
+  def test_lessons_in_class_assignments_association
+    course = Course.create(name: "Math", course_code: "AAA0890111")
+    assignment = Assignment.create( name: "Test1", course_id: course.id, percent_of_grade: 100.00)
+    lesson_assignment = Lesson.create(in_class_assignment_id: assignment.id)
+    assert lesson_assignment.in_class_assignments == assignment
+  end
+
+  def test_lessons_pre_class_assignments_association
+    course = Course.create(name: "Math", course_code: "AAA15349811")
+    assignments = Assignment.create(name: "Test2", course_id: course.id, percent_of_grade: 100.00)
+    lesson_assignments = Lesson.create(name: "2 Infinity and Beyond!!!!!!!!!!!",pre_class_assignment_id: assignments.id)
+    assert lesson_assignments.pre_class_assignments == assignments
+  end
+
+  def test_course_readings_association
+    course = Course.create(name: "CyberMen", course_code:"Error404")
+    lesson = Lesson.create(name: "Delete",course_id: course.id)
+    assert affirm(course.lessons,  'Lesson')
+    assert affirm(lesson.readings, 'Reading')
+    assert affirm(course.readings, 'Reading')
+  end
+
+  def test_school_has_name
+    school = School.create(name: "")
+    refute school.save
+  end
+
+  def test_terms_have_name_starts_on_ends_on_school_id
+    term = Term.create(name: "",starts_on: "01-01-0001",ends_on: "01-02-0001",school_id: "")
+    refute term.save
+  end
+
+  def test_user_has_name_and_email
+    user = User.create(first_name: "", last_name: "", email: "")
+    refute user.save
+  end
+
+  def test_user_email_unique
+    user = User.create(first_name:  "Test",   last_name: "User",email: "test@user.com")
+    assert user.save
+    userB = User.create(first_name: "Testing",last_name: "User",email: "test@user.com")
+    refute userB.save
+  end
+
+  def test_user_email_proper_format
+    user = User.create(first_name: "Test",last_name: "User",email: "a")
+    refute user.save
+  end
+
+  def test_user_photo_url
+      user = User.create(first_name: "Test",  last_name: "User", email: "test@users.com", photo_url: "htt://git.com")
+      refute user.save
+      user2 = User.create(first_name: "Testy",last_name: "Usery",email: "testy@usery.com",photo_url: "http://git.com")
+      assert user2.save
+  end
+
+  def test_course_code_starts_with_three_letters_ends_with_three_numbers
+    course = Course.create(name: "Daleks", course_code: "EXTERMINATE")
+    refute course.save
+    course2 = Course.create(name: "Doctor Who", course_code: "TARDIS042")
+    assert course2.save
```

Comment:
This actually shouldn't have saved. `Validate that the course_code starts with three letters and ends with three numbers`